pages/api/user/[slug].js

```diff
@@ -0,0 +1,71 @@
+import { getUserByQuery, updateUserByQuery } from "../../../lib/db-helper"
+import { isEmpty, isDefined } from "../../../utils/checks"
+
+const handler = async (req, res) => {
+  if (req.method === "GET") {
+    const { slug } = req.query
+
+    if (!isDefined(slug) || isEmpty(slug)) {
+      return res.status(404).json({
+        message: "Not found",
+        error: "No slug provided",
+      })
+    }
+
+    const [user] = await getUserByQuery(
+      { slug },
+      { email: 0, password: 0, equipment: 0, _id: 0 }
+    )
+
+    if (!user) {
+      return res.status(404).json({
+        message: "Not found",
+        error: "User not found",
+      })
+    }
+
+    return res.status(200).json(user)
+  }
+
+  if (req.method === "PATCH") {
+    const { slug } = req.query
+
+    if (!isDefined(slug) || isEmpty(slug)) {
+      return res.status(404).json({
+        message: "Not found",
+        error: "No slug provided",
+      })
+    }
+
+    const [user] = await getUserByQuery({ slug }, { email: 0, password: 0 })
+
+    if (!user) {
+      return res.status(404).json({
+        message: "Not found",
+        error: "User not found",
+      })
+    }
+
+    const [alreadyExists] = await getUserByQuery({ slug: req.body.slug })
+    if (alreadyExists) {
+      return res.status(409).json({
+        message: "Conflict",
+        error: "Slug already exists",
+      })
+    }
+
+    const update = {
+      slug: req.body.slug,
+    }
+
+    await updateUserByQuery({ _id: user._id }, update)
+    return res.status(200).json(user)
```

Comment:
Please update the user based on the session instead of using the req.query. Otherwise, unauthorized people would be able to update other people's slug using the API.



```
const session = await getServerSession(req, res, authOptions)
await updateUserByQuery({ email: session.user.email }, update)
```

Reply:
woops, sure

Reply:
this should be fixed in [this commit](https://github.com/Vincenius/workout-lol/pull/24/commits/6b0ad99d3048e38027cf83f8de6cb4a79886ad2e)